lib/src/solver/report.dart

```diff
@@ -130,6 +130,40 @@ class SolveReport {
     }
   }
 
+  /// Displays a two-line message, number of outdated packages and an
+  /// instruction to run `pub outdated` if outdated packages are detected.
+  void reportOutdated() {
+    var outdatedPackagesCount = 0;
+
+    var names = _result.packages.map((id) => id.name).toList();
+    for (var name in names) {
+      var newId = _dependencies[name];
+
+      // See if there are any newer versions of the package that we were
+      // unable to upgrade to.
+      if (newId != null) {
+        var versions = _result.availableVersions[newId.name];
+
+        for (var version in versions) {
+          if (version > newId.version) {
+            ++outdatedPackagesCount;
+            break;
+          }
+        }
```

Comment:
This can be done with something like `versions.any((v) => v > newId.version)`

But also if there is a newer version and the newer version is a pre-release we don't want to count it. Well, unless the current version is also a pre-release.

Reply:
Done!